cpp/open3d/geometry/PointCloud.cpp

```diff
@@ -478,6 +478,41 @@ std::shared_ptr<PointCloud> PointCloud::RandomDownSample(
     return SelectByIndex(indices);
 }
 
+std::shared_ptr<PointCloud> PointCloud::FarthestPointDownSample(
+        size_t num_samples) const {
+    if (num_samples == 0) {
+        return std::make_shared<PointCloud>();
+    } else if (num_samples == points_.size()) {
+        return std::make_shared<PointCloud>(*this);
+    } else if (num_samples > points_.size()) {
+        utility::LogError(
+                "Illegal number of samples: {}, must <= point size: {}",
+                num_samples, points_.size());
+    } else {
+        std::vector<size_t> indices;
+        indices.resize(num_samples);
```

Comment:
`reserve()` then `push_back()` [is faster](https://stackoverflow.com/questions/1461276/stdvector-reserve-and-push-back-is-faster-than-resize-and-array-index-w).